auth-server/src/test/java/com/makeurpicks/service/OAuthServiceTest.java

```diff
@@ -0,0 +1,199 @@
+package com.makeurpicks.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.contains;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestDatabase;
+import org.springframework.security.oauth2.provider.ClientDetails;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.makeurpicks.AuthServerApplication;
+import com.makeurpicks.dao.ClientDetailsDAO;
+import com.makeurpicks.domain.OAuthClientDetails;
+import com.makeurpicks.domain.OAuthClientDetailsBuilder;
+import com.makeurpicks.exception.OAuthclientValidationException;
+import com.makeurpicks.exception.OAuthclientValidationException.OAuthClientExceptions;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = AuthServerApplication.class)
+//@RunWith(MockitoJUnitRunner.class)
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class OAuthServiceTest {
+
+	
+	@Autowired
+	private OAuthClientsService oAuthClientService;
+	
+	
+	
+	@Mock
+	public ClientDetailsDAO clientDetailsDAO;
+	
+
+	@Rule
+	public ExpectedException expectedEx = ExpectedException.none();
+	
+	@Before
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+		
+		OAuthClientDetails authClientDetails=new OAuthClientDetails();
+		authClientDetails.setClientId("1");
+		authClientDetails.setClient_secret("111111");
+		
+		OAuthClientDetails authClientDetails2=new OAuthClientDetails();
+		authClientDetails.setClientId("2");
+		authClientDetails.setClient_secret("22222");
+		clientDetailsDAO.save(authClientDetails);
+		clientDetailsDAO.save(authClientDetails2);
+	}
+	
+	@Test
+	public void testGetAllClients(){
+		if(oAuthClientService==null){
+			fail();
+		}
+		Iterable<OAuthClientDetails > clients=oAuthClientService.findAllClients();
```

Comment:
Hi Powell,
It has been fixed to execute with proper stubs. Please check now.

Thanks
Hari